mypy/semanal.py

```diff
@@ -4381,3 +4410,21 @@ def find_fixed_callable_return(expr: Expression) -> Optional[CallableType]:
             if isinstance(t.ret_type, CallableType):
                 return t.ret_type
     return None
+
+
+def mark_any_non_explicit(t: Type) -> None:
+    """For all types within t, if that type is Any, set explicit to False"""
+    t.accept(MarkAnyNonExplicit())
+
+
+class MarkAnyNonExplicit(TypeQuery[None]):
```

Comment:
Hmm actually that might not be a good idea, since mutating types is only safe if you are certain that there is ever only a single reference to the type object. What about a type transform that replaces `Any` instances with a copy that has `explicit` set to `False`? You should be able to use `TypeTranslator`.